### PR TITLE
cmd: change default mimetype to empty string

### LINKED ISSUE #181: 24X, 25X, 27X formats cannot be downloaded
If you run this command:

~~~
youtubedr info https://www.youtube.com/watch?v=ADFgbePK7gw
~~~

Result:

~~~
ITAG | FPS |  VIDEO  |  AUDIO 
     |     | QUALITY | QUALITY
-----+-----+---------+--------
  18 |  25 | 360p    | low    
 133 |  25 | 240p    |        
 134 |  25 | 360p    |        
 135 |  25 | 480p    |        
 136 |  25 | 720p    |        
 137 |  25 | 1080p   |        
 140 |   0 |         | medium 
 160 |  25 | 144p    |        
 242 |  25 | 240p    |        
 243 |  25 | 360p    |        
 244 |  25 | 480p    |        
 247 |  25 | 720p    |        
 248 |  25 | 1080p   |        
 249 |   0 |         | low    
 250 |   0 |         | low    
 251 |   0 |         | medium 
 278 |  25 | 144p    |        
~~~

However any format after `160`, cannot be downloaded:

~~~
PS C:\> youtubedr download -q 242 https://www.youtube.com/watch?v=ADFgbePK7gw
2021/04/28 15:20:46 GET
https://youtube.com/get_video_info?video_id=ADFgbePK7gw&eurl=https://youtube.googleapis.com/v/ADFgbePK7gw
unable to find format with itag 242
~~~

### PR DESCRIPTION
**Pull request by 89z**
Fixes the error introduced by @itscaro in PR #167

Fixes #181

### CODE REVIEW

**Comment by itscaro** [comment]
This should be modified too: https://github.com/89z/youtube/blob/master/downloader/downloader.go#L118-L121

**Comment by 89z** [comment]
New commit pushed

File: downloader/downloader.go [hunk 1]
```diff
@@ -115,11 +115,6 @@ func getVideoAudioFormats(v *youtube.Video, quality string, mimetype string) (*y
 	var videoFormat, audioFormat *youtube.Format
 	var videoFormats, audioFormats youtube.FormatList
 
-	// Default mime-type to mp4
-	if mimetype == "" {
-		mimetype = "mp4"
-	}
-
 	formats := v.Formats.Type(mimetype)
```
Comment (itscaro) [inline downloader/downloader.go:118]:
```suggestion

	formats := v.Formats
	if mimetype == "" {
		formats := formats.Type(mimetype)
	}
```

Comment (89z) [reply downloader/downloader.go:118]: shouldnt that be `!=`? What is your problem dude?

**Comment by itscaro** [comment]
mimetype was set to "mp4" to keep the behavior of `hd*` quality https://github.com/kkdai/youtube/pull/167/files#diff-0560d473838290dc1f7cb1ae6dce590e1e202d9c5dea4bc077c4e4cc98541f93L62-L75

Due to the removal of pinning an itag to a quality label, mime-type is needed. Without mime-type video / audio could be mixed up in `hd*` cases

I think we should set mime-type here https://github.com/kkdai/youtube/blob/210ff51a69ced3ff4324f260495285b29f48f6df/downloader/downloader.go#L60-L61, to ensure that in composite download the same codec should be chosen?

**Comment by 89z** [comment]
@itscaro I dont care the reasons, that pull request should never have been commited. Sloppy work. You should go back to the drawing board and reimplement it, if you need it.

File: downloader/downloader.go (hunk 1 above)
Comment (itscaro) [reply downloader/downloader.go]: yes, I wanted to say that there are more changes needed around this part, I just have a quick look at your PR.

**Comment by 89z** [comment]
I removed my previous comment. I want to add some more context, since I see @itscaro reacted to it. @itscaro, this is not your personal repo, this is a repo used by many people. You have a responsibility to take care into the Pull Request you make.

So when I see a Pull Request like #167, where you killed half the formats (see #181), that should not happen. Even basic testing would have caught that.

**Comment by itscaro** [comment]
I do have a modifed version of this repo and I used it not this repo. PR is proposed and I did not merge. I took responsibility, but as I said I miss some use-case, I'm sorry about that. Error happens.

**Comment by 89z** [comment]
I have deleted my fork, as I intend to start a new repo with the same name. If any further changes are needed, they will have to be done manually.

**Comment by Julian-Chu** [comment]
@89z  , thanks for your PR.
this is a very interesting situation, two bug fix for same issue, both can work.
I will take #183 ,  because of the following reasons:
- keeping default mime type is better
- should check and handle mime type earlier (later I will remove mime type check from `getVideoAudioFormats`)

something we have learned from your PR, thanks again for your contribution.

**Pull request closed without merging**